File: modules/module_opensource/src/main/java/com/example/william/my/module/opensource/adapter/SwipeRecyclerAdapter.java
package com.example.william.my.module.opensource.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.daimajia.swipe.SwipeLayout;
import com.daimajia.swipe.adapters.RecyclerSwipeAdapter;
import com.example.william.my.module.opensource.R;

import java.util.List;

public class SwipeRecyclerAdapter extends RecyclerSwipeAdapter<SwipeRecyclerAdapter.ViewHolder> {

    private List<String> mData;

    public SwipeRecyclerAdapter(List<String> mData) {
        this.mData = mData;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.open_item_swipe, parent, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, final int i) {
        if (viewHolder.itemView instanceof SwipeLayout) {
            mItemManger.bindView(viewHolder.itemView, i);
        }
        viewHolder.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mItemManger.closeItem(i);
                mItemManger.closeAllItems();
            }
        });
        viewHolder.textView.setText(mData.get(i));
    }

    @Override
    public int getItemCount() {
        return mData == null ? 0 : mData.size();
    }

    @Override
    public int getSwipeLayoutResourceId(int position) {
        return R.id.item_swipe_swipeLayout;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private final Button button;
        private final TextView textView;

        public ViewHolder(View itemView) {
            super(itemView);
            button = itemView.findViewById(R.id.item_swipe_button);
            textView = itemView.findViewById(R.id.item_swipe_textView);
        }
    }

    public void setData(List<String> data) {
        this.mData = data;
    }
}

File: modules/module_opensource/src/main/java/com/example/william/my/module/opensource/adapter/SwipeRecyclerAdapter.kt
package com.example.william.my.module.opensource.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.daimajia.swipe.SwipeLayout
import com.daimajia.swipe.adapters.RecyclerSwipeAdapter
import com.example.william.my.module.opensource.R
import com.example.william.my.module.opensource.databinding.OpenItemSwipeBinding

class SwipeRecyclerAdapter(private var data: List<String>?) :
    RecyclerSwipeAdapter<SwipeRecyclerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding =
            OpenItemSwipeBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        if (viewHolder.itemView is SwipeLayout) {
            mItemManger.bindView(viewHolder.itemView, i)
        }

        viewHolder.binding.itemSwipeButton.setOnClickListener {
            mItemManger.closeItem(i)
            mItemManger.closeAllItems()
        }
        data?.let {
            viewHolder.binding.itemSwipeTextView.text = it[i]
        }
    }

    override fun getItemCount(): Int {
        return data?.size ?: 0
    }

    override fun getSwipeLayoutResourceId(position: Int): Int {
        return R.id.item_swipe_swipeLayout
    }

    class ViewHolder(val binding: OpenItemSwipeBinding) :
        RecyclerView.ViewHolder(binding.root)
}
